EIPS/eip-1630.md

```diff
@@ -0,0 +1,265 @@
+---
+eip: 1630
+title: Hashed Time-Locked Contracts
+author: Matthew Black <matthew.black@consensys.net>, TingWei Liu <ting.liu@consensys.net>, Liquality Team <info@liquality.io>
+status: Draft
+discussions-to: https://github.com/ethereum/EIPs/issues/1631
+type: Standards Track
+category: ERC
+created: 2018-11-28
+---
+
+## Simple Summary
+
+A standard EVM script for generalized payments that acknowledges receiving the payment prior to a deadline.
+
+## Abstract
+
+A Hashed Time-Lock Contract (HTLC) is a script that permits a designated party (the "seller") to spend funds by disclosing the preimage of a hash. It also permits a second party (the "buyer") to spend the funds after a timeout is reached, in a refund situation.
+
+## Motivation
+
+HTLC transactions are a safe and cheap method of exchanging secrets for money over the blockchain, due to the ability to recover funds from an uncooperative counterparty, and the opportunity that the possessor of a secret has to receive the funds before such a refund can occur.
+
+HTLC's enable cross-chain [atomic swaps](https://en.bitcoin.it/wiki/Atomic_swap)
+
+## Definitions
+
+`msg.sender`: is always the address where the current (external) function call came from.
+`buyer`: entity that receives funds from `seller` once the `seller` reveals the `secret`
+`seller`: entity that contributes funds to the `buyer` by revealing the `secret` or refunds after `expiration`
+`secret`: random number chosen by the `seller`, revealed to allow the `buyer` to redeem the funds
+`secretHash`: hash of the `secret`, used in the construction of HTLC
+`expiration`: timestamp the determines when `seller` and `buyer` can redeem
+`now`: current block timestamp
+
+## Specification
+
+### Constructor
+
+The `msg.sender`, transfers funds to the smart contract while deploying
+
+```js
+constructor (bytes32 _secretHash, uint256 _expiration, address _buyer) public payable {
```

Comment:
If you are using Solidity syntax here, perhaps just close it off with a semicolon and not curly brackets.